internal/expr/query.go

```diff
@@ -132,19 +181,43 @@ func (qe *QueryExpr) ScanArgs(columns []string, outputArgs []any) ([]any, error)
 			continue
 		}
 		if idx >= len(qe.outputs) {
-			return nil, fmt.Errorf("internal error: sqlair column not in outputs (%d>=%d)", idx, len(qe.outputs))
-		}
-		field := qe.outputs[idx]
-		outputVal, ok := typeDest[field.structType]
-		if !ok {
-			return nil, fmt.Errorf("type %q found in query but not passed to decode", field.structType.Name())
+			return nil, nil, fmt.Errorf("internal error: sqlair column not in outputs (%d>=%d)", idx, len(qe.outputs))
 		}
+		typeElement := qe.outputs[idx]
+		switch te := typeElement.(type) {
+		case field:
+			outputVal, ok := typeDest[te.structType]
+			if !ok {
+				return nil, nil, fmt.Errorf("type %q found in query but not passed to decode", te.structType.Name())
+			}
 
-		val := outputVal.FieldByIndex(field.index)
-		if !val.CanSet() {
-			return nil, fmt.Errorf("internal error: cannot set field %s of struct %s", field.name, field.structType.Name())
+			val := outputVal.FieldByIndex(te.index)
+			if !val.CanSet() {
+				return nil, nil, fmt.Errorf("internal error: cannot set field %s of struct %s", te.name, te.structType.Name())
+			}
+			ptrs = append(ptrs, val.Addr().Interface())
+		case mapKey:
+			v, ok := m[te.name]
+			// If there is already an value in the map for this column name, scan into a variable of that type.
+			if ok {
+				val := reflect.New(reflect.TypeOf(v)).Elem()
+				addr := val.Addr().Interface()
+				ptrs = append(ptrs, addr)
+				mapPtrs = append(mapPtrs, addr)
+			} else {
+				var y any
+				ptrs = append(ptrs, &y)
+				mapPtrs = append(mapPtrs, &y)
+			}
+			keyIndex[te.name] = len(mapPtrs) - 1
 		}
-		ptrs = append(ptrs, val.Addr().Interface())
 	}
-	return ptrs, nil
+	return ptrs, &MapDecodeInfo{mapPtrs: mapPtrs, keyIndex: keyIndex, m: m}, nil
+}
+
+// PopulateMap enters the scanned values into the map.
+func (mapDecodeInfo *MapDecodeInfo) PopulateMap() {
```

Comment:
This is really small, but somehow 'puts' feels more natural to me? Also should the 'the' be removed?

Will `PopluateMap` is the only method on this type, perhaps just `Populate` would be cleaner?

Reply:
Changed.